### PR TITLE
Support Version 2.0.0 changes for update-fc

### LINKED ISSUE #214: Support Version 2.0.0 changes for update-fc
1. Generate server stub for the version 2.0.0 specification 
2. If the api/openapi.yaml is not updated as per the published version 2.0.0 , then copy the api/openapi.yaml file from the generated stub to the source code
3. no change required in the method signature

- [ ] In the version 1 implementation , this information is included in the configuration file itself. But , in the version 2 , this shall be included to the application data (i.e) elastic search. [ElasticSearch://core-model-1-4:network-control-domain/control-construct=*/forwarding-domain=*-op-fd-000/forwarding-construct=*-op-fc-*/uuid]

### PR DESCRIPTION
**Pull request by vanithavalluripalli9**
Fixes #214

### CODE REVIEW

File: server/service/IndividualServicesService.js [hunk 1]
```diff
@@ -1159,36 +1162,17 @@ exports.updateFc = function (body, user, originator, xCorrelator, traceIndicator
       let controlConstructUuid = figureOutControlConstructUuid(forwardingConstructUuid);
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:1162]: You can find correct control construct by searching in ES for forwardingConstructUuid.

File: server/service/IndividualServicesService.js [hunk 2]
```diff
@@ -44,6 +44,9 @@ const Link = require('./models/Link');
 const { elasticsearchService, getIndexAliasAsync } = require('onf-core-model-ap/applicationPattern/services/ElasticsearchService');
 const ControlConstructService = require('./individualServices/ControlConstructService');
 
+const ELASTICSEARCH_CLIENT_CC_UUID = "alt-2-0-1-es-c-es-1-0-0-000";
+const ELASTICSEARCH_CLIENT_LINKS_UUID = "alt-2-0-1-es-c-es-1-0-0-001";
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:48]: Not used.

File: server/service/IndividualServicesService.js [hunk 3]
```diff
@@ -44,6 +44,9 @@ const Link = require('./models/Link');
 const { elasticsearchService, getIndexAliasAsync } = require('onf-core-model-ap/applicationPattern/services/ElasticsearchService');
 const ControlConstructService = require('./individualServices/ControlConstructService');
 
+const ELASTICSEARCH_CLIENT_CC_UUID = "alt-2-0-1-es-c-es-1-0-0-000";
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:47]: Given the amount of methods, I'd create 'ForwardingService' that would group methods dealing with FC and FC-ports and not pollute IndividualServices.

File: server/service/IndividualServicesService.js [hunk 4]
```diff
@@ -1377,6 +1361,102 @@ function getAllClientApplicationList() {
   });
 }
 
+async function updateForwardingConstrutList(forwardingConstructListToBeUpdated, forwardingConstructUuid) {
+  return new Promise(async function (resolve, reject) {
+    try {
+      let client = await elasticsearchService.getClient(false, "ELASTICSEARCH_CLIENT_CC_UUID");
+      let indexAlias = await getIndexAliasAsync("ELASTICSEARCH_CLIENT_CC_UUID");
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:1368]: Value should be used here, not the name of the constant.

File: server/service/IndividualServicesService.js [hunk 5]
```diff
@@ -1377,6 +1361,102 @@ function getAllClientApplicationList() {
   });
 }
 
+async function updateForwardingConstrutList(forwardingConstructListToBeUpdated, forwardingConstructUuid) {
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:1364]: typo in the name of the method -> updateForwardingConstrucTList

File: server/service/IndividualServicesService.js [hunk 6]
```diff
@@ -1377,6 +1361,102 @@ function getAllClientApplicationList() {
...
+
+async function getForwardingDomainFromControlConstruct(controlConstructUuid) {
+  return new Promise(async function (resolve, reject) {
+    let forwardingDomainOfControlConstruct = {}
+    try {
+      let client = await elasticsearchService.getClient(false, ELASTICSEARCH_CLIENT_CC_UUID);
+      let indexAlias = await getIndexAliasAsync(ELASTICSEARCH_CLIENT_CC_UUID);
+      let res = await client.search({
+        index: indexAlias,
+        filter_path: "hits.hits._id,hits.hits._source.forwarding-domain",
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:1401]: Given how this method is used, you want actually the forwardingConstructList, you can filter it in the request.

File: server/service/IndividualServicesService.js [hunk 7]
```diff
@@ -1377,6 +1361,102 @@ function getAllClientApplicationList() {
...
+      if (Object.keys(forwardingConstructListToBeUpdated).length >= 2) {
+
+        response = await client.update({
+          index: indexAlias,
+          id: forwardingConstructToBeUpdated.documentId,
+          body: {
+            script: {
+              source: "ctx._source['forwarding-domain'][0]['forwarding-construct'] = params['forwardingConstructList']",
+              params: {
+                "uuid": forwardingConstructUuid,
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:1379]: not used

**Review by DanaSunal — CHANGES_REQUESTED** [review]

File: server/service/IndividualServicesService.js (hunk 1 above)
Comment (vanithavalluripalli9) [reply server/service/IndividualServicesService.js:1161]: There are scenarios where forwardingConstructUuid might not be present in controlconstruct and we have to add it . So, got the control construct and checked whether the forwardingConstructUuid exists if not added sand if present replaced

File: server/service/IndividualServicesService.js (hunk 2 above)
Comment (vanithavalluripalli9) [reply server/service/IndividualServicesService.js]: Removed

File: server/service/IndividualServicesService.js (hunk 3 above)
Comment (vanithavalluripalli9) [reply server/service/IndividualServicesService.js]: Created the ForwardinService file as suggested and moved the methods

File: server/service/IndividualServicesService.js (hunk 6 above)
Comment (vanithavalluripalli9) [reply server/service/IndividualServicesService.js]: Yes tried it . But , was not able to filter only forwardingConstructList . So, fetched forwarding-domain as it didn't make much difference. If you could suggest a way is ES to fetch only forwardingConstructList will try it .Thank you

File: server/service/IndividualServicesService.js (hunk 7 above)
Comment (vanithavalluripalli9) [reply server/service/IndividualServicesService.js]: Done.

File: server/service/IndividualServicesService.js (hunk 6 above)
Comment (DanaSunal) [reply server/service/IndividualServicesService.js]: filter_path: "hits.hits._id,hits.hits._source.forwarding-domain.forwarding-construct"

Comment (vanithavalluripalli9) [reply server/service/IndividualServicesService.js]: Thank you . Tried this before but output was something like forwarding-domain:[{forwarding-construct}] which didn't make much difference than the current one .

**Review by DanaSunal — APPROVED** [review]

**Pull request merged**